distributed/scheduler.py

```diff
@@ -528,7 +572,9 @@ def __repr__(self):
             len(self._processing),
         )
 
-    def identity(self):
+    @ccall
+    @exceptval(check=False)
+    def identity(self) -> dict:
```

Comment:
As we are annotating the `return` with a Python object and using `ccall`, Cython gets a bit confused. Cython typically wants to add exception handling code when return a non-Python object (like `double`), but this isn't needed with Python objects as these are pointers, which will be set to `NULL` on an exception. Cython already checks for this afterwards (as it does with all Python objects). Hence we add `exceptval(check=False)` to stop Cython from adding this extra code, which fixes the issue.